src/nupic/research/temporal_memory.py

```diff
@@ -120,394 +222,200 @@ def compute(self, activeColumns, learn=True):
-                            -self.predictedSegmentDecrement,
-                            0.0)
+    return cellsToAdd
 
-
-
-  def computePredictiveCells(self, activeCells, connections):
+  @staticmethod
+  def burstColumn(cellsPerColumn, columnDimensions, connections, excitedColumn,
+                  learn, initialPermanence, matchingSegments, maxNewSynapseCount,
+                  permanenceDecrement, permanenceIncrement, prevActiveCells,
+                  prevWinnerCells, random):
     """
-    Phase 4: Compute predictive cells due to lateral input
-    on distal dendrites.
-
-    Pseudocode:
-
-      - for each distal dendrite segment with activity >= activationThreshold
-        - mark the segment as active
-        - mark the cell as predictive
-
-      - if predictedSegmentDecrement > 0
-        - for each distal dendrite segment with unconnected
-          activity >=  minThreshold
-          - mark the segment as matching
-          - mark the cell as matching
-
-    Forward propagates activity from active cells to the synapses that touch
-    them, to determine which synapses are active.
-
-    @param activeCells (set)         Indices of active cells in `t`
-    @param connections (Connections) Connectivity of layer
+    @param excitedColumn   (dict)  Excited Column instance from excitedColumnsGenerator
+    @param learn           (bool)  Whether or not learning is enabled
+    @param prevActiveCells (list)  Active cells in `t-1`
+    @param prevWinnerCells (list)  Winner cells in `t-1`
 
     @return (tuple) Contains:
-                      `activeSegments`  (set),
-                      `predictiveCells` (set),
-                      `matchingSegments` (set),
-                      `matchingCells`    (set)
-    """
-    numActiveConnectedSynapsesForSegment = defaultdict(int)
-    numActiveSynapsesForSegment = defaultdict(int)
-    activeSegments = set()
-    predictiveCells = set()
-
-    matchingSegments = set()
-    matchingCells = set()
-
-    for cell in activeCells:
-      for synapseData in connections.synapsesForPresynapticCell(cell).values():
-        segment = synapseData.segment
-        permanence = synapseData.permanence
-
-        if permanence >= self.connectedPermanence:
-          numActiveConnectedSynapsesForSegment[segment] += 1
-
-          if (numActiveConnectedSynapsesForSegment[segment] >=
-              self.activationThreshold):
-            activeSegments.add(segment)
-            predictiveCells.add(connections.cellForSegment(segment))
-
-        if permanence > 0 and self.predictedSegmentDecrement > 0:
-          numActiveSynapsesForSegment[segment] += 1
-
-          if numActiveSynapsesForSegment[segment] >= self.minThreshold:
-            matchingSegments.add(segment)
-            matchingCells.add(connections.cellForSegment(segment))
+                      `cells`         (list),
+                      `bestCell`      (int),
+    """
+    cells = []
```

Comment:
My bad. This needs to get populated in both paths, since it gets returned by the method.